### PR TITLE
Makes bluespace navigation gigabeacons both significantly nicer to use, and significantly cooler

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

"Behold, a navigation beacon" Plato says to the crowd of shocked onlookers.

image

What are you looking at here? Navigation beacons have been party reworked for significantly improved ease of use, and for significantly cooler functionality.

As it stands, in order to explore space with a ship of some sort, you have to take the following steps:

Fly to target zlevel
Build a floor
Build a machine frame, then make the beacon out of it
Go back and park your ship

Now, while this doesn't eliminate this process, it makes it a step shorter and a good bit less annoying to do. There are no gigabeacon boards, instead a compact beacon can be printed. This can be used inhand and, after a delay of three seconds, will create a brand new gigabeacon in front of you with no extra hassle.

So, aside from the sprite (the old one is an alien power core, by the way), what's new? Just a bit of flavor. Examining a gigabeacon will mention a radio frequency. If you tune into this radio frequency, then you might be able to catch the automatic messages the beacons put out every five minutes. They detail a little bit about the beacon, namely its coordinates, and if its activate or locked. While not critical info my any means, it can be good information to have pop up every few minutes, or for discovering that there are gigabeacons you didn't know were there before.
## Why It's Good For The Game

Gigabeacons had two big problems, one was that they were pointlessly annoying to set up, and two was that they didn't have their own sprite. Yup, they just used a random alien tech structure and nobody said anything about it. Hopefully this pr fixes both of those problems, alongside adding a little neat bit of flavor to them in the form of the automatic radio broadcasts they put every few minutes on a non-standard channel.
## Changelog
:cl:
qol: Gigabeacons no longer need to be constructed like standard machines, instead being printed as items that can be deployed quickly inhand without tools.
qol: Gigabeacons are no longer their own research node, instead being a part of telecomms tech (due to the fact they're RADIO navigation beacons)
image: Navigation gigabeacons now have their own unique sprites, inspired by maritime EPIRB devices. If you find a yellow box with an antenna on it, you probably found a nav beacon!
code: The deployable component now gives you an option to disable the examine blurb it adds to its parent
/:cl:

### CODE REVIEW

**Comment by newfren** [comment]
how large are these when they aren't deployed? extremely great pr

**Comment by Paxilmaniac** [comment]
> 

Normal sized, you could fit a few in a backpack if you wanted to

**Comment by atlasle** [comment]
oh so thats what those do, great pr

**Comment by Acer202** [comment]
Looks like some sort of maritime distress/signal buoy in space and I'm all for it! Nice.

**Comment by Paxilmaniac** [comment]
> Looks like some sort of maritime distress/signal buoy in space and I'm all for it! Nice.

That was the inspiration, yeah

**Comment by AdamElTablawy** [comment]
Problem: engiborgs with upgrades can’t build gigabeacons like this now. 

A rare but circumstantial problem, in the event that like; a rogue engiborg wants to build a nab beacon to move the AI’s stolen cyborg mothership or something off zlevel, or to make an entrenched teleport beacon to lure people into a trap etc 

Or it was just helping with space exploration

**Comment by atlasle** [comment]
> Problem: engiborgs with upgrades can’t build gigabeacons like this now. 

Could be fixed by making it so clicking on the item with like a multitool does the same thing as using it in hand, so engiborgs could pull them behind themselves

**Review by Wallemations — APPROVED** [review]

File: code/datums/components/deployable.dm [hunk 1]
```diff
@@ -17,17 +17,21 @@
 	var/deployed_name
 	/// If the item used to deploy gets deleted on use or not
 	var/delete_on_use = TRUE
+	/// If the component adds a little bit into the parent's description
+	var/add_description_hint = TRUE
 
-/datum/component/deployable/Initialize(deploy_time, thing_to_be_deployed, delete_on_use)
+/datum/component/deployable/Initialize(deploy_time, thing_to_be_deployed, delete_on_use, add_description_hint)
```
Comment (ZephyrTFA) [inline code/datums/components/deployable.dm:23]: You need to add a default to this

File: code/datums/components/deployable.dm [hunk 2]
```diff
@@ -17,17 +17,21 @@
...
 	. = ..()
 	if(!isitem(parent))
 		return COMPONENT_INCOMPATIBLE
 
 	src.deploy_time = deploy_time
 	src.thing_to_be_deployed = thing_to_be_deployed
 	src.delete_on_use = delete_on_use
+	src.add_description_hint = add_description_hint
```
Comment (ZephyrTFA) [inline code/datums/components/deployable.dm:31]: or check for it being null before setting it

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

**Pull request merged**